src/abstractions/software/assembler/symbolTable/index.js

```diff
@@ -0,0 +1,56 @@
+import { getPredefinedSymbols } from './utils'
+
+class SymbolTable {
+  /**
+   * creates a symbol table and initialize it with the predifined symbols
+   */
+  constructor () {
+    this.table = {
+      ...getPredefinedSymbols()
+    }
+  }
+
+  /**
+   * @param {string} symbol symbol
+   * @param {number} address address
+   * Add  (symbol, address) pair to the table
```

Comment:
The function description should be above the param descriptions, right?
```suggestion
   * Add  (symbol, address) pair to the table
   * @param {string} symbol symbol
   * @param {number} address address of the symbol
```